runtime/src/accounts_index.rs

```diff
@@ -366,6 +383,8 @@ impl<T: 'static + Clone> AccountsIndex<T> {
         F: FnMut(&Pubkey, (&T, Slot)),
         R: RangeBounds<Pubkey>,
     {
+        // TODO: expand to use mint index to find the `pubkey_list` below more efficiently
+        // instead of scanning the entire range
```

Comment:
@CriesofCarrots  have to expand this method to leverage the new index for scans that query by the mint